runtime/permissions/lib.rs

```diff
@@ -689,11 +689,141 @@ impl Descriptor for WriteDescriptor {
+
+    // Retrieve the port from the URL, if specified, or use the default port for the scheme
+    let port = url.port_or_known_default();
+    // Return the constructed NetDescriptor
+    Ok(NetDescriptor(host.to_string(), port))
+  }
+
+  /// Parses a string representation of a network descriptor, extracting the host and optional port.
+  ///
+  /// This function handles various input formats including URLs, IPv4, and IPv6 addresses.
+  /// It attempts to extract the host and port components, validating them as necessary.
+  ///
+  /// # Examples
+  /// /*None: its goes on default port*/
+  ///
+  /// ```
+  /// // Parsing host address without a port
+  /// use deno_permissions::NetDescriptor;
+  /// assert_eq!(NetDescriptor::from_str("deno.land").unwrap(), NetDescriptor("deno.land".to_string(), None));
+  /// // Parsing host address with a port
+  /// assert_eq!(NetDescriptor::from_str("deno.land:80").unwrap(), NetDescriptor("deno.land".to_string(), Some(80)));
+  ///
+  /// // Parsing an IPv4 address
+  /// assert_eq!(NetDescriptor::from_str("127.0.0.1").unwrap(), NetDescriptor("127.0.0.1".to_string(), None));
+  /// // Parsing an IPv4 address with a port
+  /// assert_eq!(deno_permissions::NetDescriptor::from_str("127.0.0.1:80").unwrap(), NetDescriptor("127.0.0.1".to_string(), Some(80)));
+  ///
+  /// // Parsing an IPv6 address
+  /// assert_eq!(NetDescriptor::from_str("[2606:4700:4700::1111]").unwrap(), NetDescriptor("[2606:4700:4700::1111]".to_string(), None));
+  /// // Parsing an IPv6 address with a port
+  /// assert_eq!(NetDescriptor::from_str("[2606:4700:4700::1111]:80").unwrap(), NetDescriptor("[2606:4700:4700::1111]".to_string(), Some(80)));
+  ///
+  /// // Parsing a URL with a host
+  /// assert_eq!(NetDescriptor::from_str("https://github.com/denoland/").unwrap(), NetDescriptor("github.com".to_string(), None));
+  /// // Parsing a URL with a host & port
+  /// assert_eq!(NetDescriptor::from_str("https://github.com:443/denoland/").unwrap(), NetDescriptor("github.com".to_string(), Some(443)));
+  ///
+  /// // Parsing a URL with an IPv4 address
+  /// assert_eq!(NetDescriptor::from_str("https://127.0.0.1").unwrap(), NetDescriptor("127.0.0.1".to_string(), None));
+  /// // Parsing a URL with an IPv4 address & port
+  /// assert_eq!(NetDescriptor::from_str("https://127.0.0.1:80").unwrap(), NetDescriptor("127.0.0.1".to_string(), Some(80)));
+  ///
+  /// // Parsing a URL with an IPv6 address
+  /// assert_eq!(NetDescriptor::from_str("https://[2606:4700:4700::1111]").unwrap(), NetDescriptor("[2606:4700:4700::1111]".to_string(), None));
+  /// // Parsing a URL with an IPv6 address & port
+  /// assert_eq!(NetDescriptor::from_str("https://[2606:4700:4700::1111]:80").unwrap(), NetDescriptor("[2606:4700:4700::1111]".to_string(), Some(80)));
+  /// ```
+  ///
+  /// # Errors
+  ///
+  /// Returns an `AnyError` if the input string cannot be parsed into a valid host and optional port.
```

Comment:
This documentation comment is a lot to maintain.

Reply:
ok